idunn/geocoder/models/params.py

```diff
@@ -0,0 +1,87 @@
+"""
+Bragi parameters as defined here:
+  https://github.com/CanalTP/mimirsbrunn/blob/master/libs/bragi/src/routes/autocomplete.rs#L59
+"""
+from enum import Enum
+from typing import List, Optional
+
+from fastapi import Query
+from pydantic import BaseModel, PositiveInt, confloat, conint
+
+from idunn import settings
+from .cosmogony import ZoneType
+
+
+class Type(str, Enum):
+    City = "city"
+    House = "house"
+    Poi = "poi"
+    StopArea = "public_transport:stop_area"
+    Street = "street"
+    Zone = "zone"
+
+
+class QueryParams(BaseModel):
```

Comment:
It is actually possible to simplify this class and keep the generated docs, thanks to Pydantic's `dataclass` decorator:  
https://pydantic-docs.helpmanual.io/usage/dataclasses/